src/components/data-management/ProjectDetails.jsx

```diff
@@ -104,6 +115,47 @@ const ProjectDetails = ({ width, height }) => {
     }
   }, [samples, activeProject]);
 
+  const downloadData = async (type) => {
+    const experimentId = activeProject?.experiments[0];
+
+    try {
+      if (!environment) throw new Error('Environment is not set');
+      if (!Object.values(downloadTypes).includes(type)) throw new Error('Invalid download type');
+
+      const { signedUrl } = await getFromApiExpectOK(`/v1/experiments/${experimentId}/download/${type}`);
+
+      const link = document.createElement('a');
+      link.style.display = 'none';
+      link.href = signedUrl;
+      link.download = `${type}_${experimentId}.rds`;
+
+      document.body.appendChild(link);
+      link.click();
+
+      setTimeout(() => {
+        URL.revokeObjectURL(link.href);
+        link.parentNode.removeChild(link);
+      }, 0);
+    } catch (e) {
+      pushNotificationMessage('error', endUserMessages.ERROR_DOWNLOADING_DATA);
+    }
+  };
+
+  useEffect(() => {
+    const { loading, error, status } = backendStatus;
+
+    if (
+      !loading
+      && !error
+      && status?.pipeline
+      && status.pipeline?.status === pipelineStatusValues.SUCCEEDED
+    ) {
+      setPipelineHasRun(true);
+    }
+
+    setPipelineHasRun(false);
```

Comment:
this is being defined on a project-level. This will break once there is more than a single-experiment per project. We are already having issues where different projects interact among them so I'd be very careful to limit the visibility of state to the relevant component.

Reply:
Refactored in 08d66f3